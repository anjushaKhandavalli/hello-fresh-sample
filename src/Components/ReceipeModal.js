import * as PropTypes from "prop-types";

const RecipeModal = ({showRecipe, onClose, foodItem}) => {
    if(!showRecipe) return null

    return(
        <div className="recipeModal" data-testid="recipeModal" onClick={onClose}>
            <div className="recipe">
                <h2>{foodItem.name}</h2>
                <h3>{foodItem.recipeData.headline}</h3>
                <h4>Plant-Based | Ready in 5 | Serves 2</h4>
                <p className="recipeInfo">{foodItem.recipeData.description}</p>
                <h5>Ingredients</h5>
                <ul className="recipeInfo">{foodItem.recipeData.ingredients.map((ingredient) => <li key={ingredient}>{ingredient}</li>)}</ul>
            </div>
        </div>
    )
}

RecipeModal.propTypes = {showRecipe: PropTypes.bool.isRequired};
export default RecipeModal;