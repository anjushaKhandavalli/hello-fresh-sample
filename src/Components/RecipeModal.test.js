import {fireEvent, getByTestId, render} from "@testing-library/react";
import ReceipeModal from "./ReceipeModal";

describe("RecipeModal", () => {
    const mockFoodItem = {
            "id": "94298",
            "imageUrl": "https://cdn.pixabay.com/photo/2021/10/10/18/04/food-6697924_1280.jpg",
            "name": "3 Chicken Wings",
            "description": "Tender, Spicy and Juicy.",
            "recipeData": {
                "description": "There\u2019s nothing like the simple things in life - the smell of freshly cut grass, sitting outside on a nice sunny day, spending time with friends and family. Well here is a recipe that delivers simple culinary pleasures - some nice fresh fish with a crispy crust, crunchy potato wedges and some delightfully sweet sugar snap peas flavoured with cooling mint. Slip into something comfortable and relax into a delicious dinner!",
                "headline": "with Sweet Potato Wedges and Minted Snap Peas",
                "ingredients": [
                    "375g Sweet Potatoes",
                    "1 Tsp Paprika",
                    "2 Tbsps Parmesan Cheese",
                    "1 Lemon",
                    "A Few Sprigs Thyme",
                    "25g Panko Breadcrumbs",
                    "1 Tbsp Butter",
                    "2 Cod Fillets",
                    "150g Sugar Snap Peas",
                    "A Few Sprigs Mint",
                    "75ml Sour Cream"
                ],
            }
        }

    it("should not render recipe modal when showRecipe is false",() => {
        const {queryByTestId} = render(<ReceipeModal showRecipe={false} foodItem={mockFoodItem} onClose={jest.fn()}/>)

        expect(queryByTestId('recipeModal')).not.toBeInTheDocument()
    })

    it("should render recipe modal when showRecipe is true",() => {
        const {getByTestId} = render(<ReceipeModal showRecipe={true} foodItem={mockFoodItem} onClose={jest.fn()}/>)

        expect(getByTestId('recipeModal')).toBeInTheDocument();
    })

    it("should render recipe modal with recipe information",() => {
        const component = render(<ReceipeModal showRecipe={true} foodItem={mockFoodItem} onClose={jest.fn()}/>)

        expect(component).toMatchSnapshot();
    })

    it("should call onClose on click of recipe modal",() => {
        const mockOnClose = jest.fn()
        const {getByTestId} = render(<ReceipeModal showRecipe={true} foodItem={mockFoodItem} onClose={mockOnClose}/>)

        fireEvent.click(getByTestId('recipeModal'))
        expect(mockOnClose).toBeCalledTimes(1);
    })
})