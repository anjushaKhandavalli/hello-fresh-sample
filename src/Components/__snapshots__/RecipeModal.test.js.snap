// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeModal should render recipe modal with recipe information 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="recipeModal"
        data-testid="recipeModal"
      >
        <div
          class="recipe"
        >
          <h2>
            3 Chicken Wings
          </h2>
          <h3>
            with Sweet Potato Wedges and Minted Snap Peas
          </h3>
          <h4>
            Plant-Based | Ready in 5 | Serves 2
          </h4>
          <p
            class="recipeInfo"
          >
            There’s nothing like the simple things in life - the smell of freshly cut grass, sitting outside on a nice sunny day, spending time with friends and family. Well here is a recipe that delivers simple culinary pleasures - some nice fresh fish with a crispy crust, crunchy potato wedges and some delightfully sweet sugar snap peas flavoured with cooling mint. Slip into something comfortable and relax into a delicious dinner!
          </p>
          <h5>
            Ingredients
          </h5>
          <ul
            class="recipeInfo"
          >
            <li>
              375g Sweet Potatoes
            </li>
            <li>
              1 Tsp Paprika
            </li>
            <li>
              2 Tbsps Parmesan Cheese
            </li>
            <li>
              1 Lemon
            </li>
            <li>
              A Few Sprigs Thyme
            </li>
            <li>
              25g Panko Breadcrumbs
            </li>
            <li>
              1 Tbsp Butter
            </li>
            <li>
              2 Cod Fillets
            </li>
            <li>
              150g Sugar Snap Peas
            </li>
            <li>
              A Few Sprigs Mint
            </li>
            <li>
              75ml Sour Cream
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="recipeModal"
      data-testid="recipeModal"
    >
      <div
        class="recipe"
      >
        <h2>
          3 Chicken Wings
        </h2>
        <h3>
          with Sweet Potato Wedges and Minted Snap Peas
        </h3>
        <h4>
          Plant-Based | Ready in 5 | Serves 2
        </h4>
        <p
          class="recipeInfo"
        >
          There’s nothing like the simple things in life - the smell of freshly cut grass, sitting outside on a nice sunny day, spending time with friends and family. Well here is a recipe that delivers simple culinary pleasures - some nice fresh fish with a crispy crust, crunchy potato wedges and some delightfully sweet sugar snap peas flavoured with cooling mint. Slip into something comfortable and relax into a delicious dinner!
        </p>
        <h5>
          Ingredients
        </h5>
        <ul
          class="recipeInfo"
        >
          <li>
            375g Sweet Potatoes
          </li>
          <li>
            1 Tsp Paprika
          </li>
          <li>
            2 Tbsps Parmesan Cheese
          </li>
          <li>
            1 Lemon
          </li>
          <li>
            A Few Sprigs Thyme
          </li>
          <li>
            25g Panko Breadcrumbs
          </li>
          <li>
            1 Tbsp Butter
          </li>
          <li>
            2 Cod Fillets
          </li>
          <li>
            150g Sugar Snap Peas
          </li>
          <li>
            A Few Sprigs Mint
          </li>
          <li>
            75ml Sour Cream
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
